ocaml/src/tools/lwt_pool2.ml

```diff
@@ -81,18 +86,35 @@ let use t f =
                end) >>= fun () ->
               Lwt.return r)
            (fun exn ->
-              if t.check a exn
-              then
+              match t.check a exn with
+              | Keep ->
                 begin
                   Queue.push a t.available_items;
                   Lwt.fail exn
                 end
-              else
+              | DropThis ->
                 begin
                   t.count <- t.count - 1;
                   t.cleanup a >>= fun () ->
                   Lwt.fail exn
-                end))
+                end
+              | DropPool ->
+                 begin
+                   t.count <- 0;
+                   t.cleanup a >>= fun () ->
+                   let rec loop () =
+                     if Queue.is_empty t.available_items
+                     then Lwt.return_unit
+                     else
+                       let a = Queue.pop t.available_items in
```

Comment:
Preferably pop all items from queue without lwt binds in between?
Currently it's hard to reason about what will happen when another thread wants to use the pool after count has been reset but before all connections have been popped from the queue...